### PR TITLE
fix: Forgot that EDGE_CHILD devices don't have DNIs

### PR DESCRIPTION
**Pull request by dljsjr**
Forgot to refactor all usages of `device_network_id` to `parent_assigned_child_key` with the previous `EDGE_CHILD` refactor.

### CODE REVIEW

File: drivers/SmartThings/philips-hue/src/handlers.lua [hunk 1]
```diff
@@ -42,13 +42,13 @@ local function do_switch_action(driver, device, args)
...
 ---@param driver HueDriver
----@param device HueDevice
+---@param device HueChildDevice
 local function do_switch_level_action(driver, device, args)
   local level = st_utils.clamp_value(args.args.level, 1, 100)
   local bridge_device = driver:get_device_info(device:get_field(Fields.PARENT_DEVICE_ID))
 
   if not bridge_device then
-    log.warn("Couldn't get a bridge for light with DNI " .. device.device_network_id)
+    log.warn("Couldn't get a bridge for light with Child Key " .. device.parent_assigned_child_key)
```
Comment (cjswedes) [inline drivers/SmartThings/philips-hue/src/handlers.lua:51]: do we need the `device.device_network_id or device.parent_assigned_child_key` here?

File: drivers/SmartThings/philips-hue/src/init.lua [hunk 2]
```diff
@@ -511,7 +511,7 @@ local function init_light(driver, device)
       device:set_field(Fields.MIN_KELVIN, HueApi.MIN_TEMP_KELVIN_WHITE_AMBIANCE, { persist = true})
     end
   end
-  local device_light_resource_id = device:get_field(Fields.RESOURCE_ID) or device.device_network_id
+  local device_light_resource_id = device:get_field(Fields.RESOURCE_ID) or device.parent_assigned_child_key
```
Comment (cjswedes) [inline drivers/SmartThings/philips-hue/src/init.lua:514]: do we need to consider old device's `device.device_network_id` here?

File: drivers/SmartThings/philips-hue/src/handlers.lua (hunk 1 above)
Comment (dljsjr) [reply drivers/SmartThings/philips-hue/src/handlers.lua:51]: Shouldn't need it, all of the handlers will only ever receive bulbs (bridges will be filtered out by now by pseudo-type checks). And lights will never have a DNI with this change.

File: drivers/SmartThings/philips-hue/src/init.lua (hunk 2 above)
Comment (dljsjr) [reply drivers/SmartThings/philips-hue/src/init.lua:514]: Yes, good eyes.

**Review by cjswedes — APPROVED** [review]

**Pull request merged**